.github/workflows/build-iso.yaml

```diff
@@ -16,12 +16,14 @@ jobs:
         run: |
           sudo apt-get update
           sudo apt-get install -y mkisofs
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
       - name: Build iso from script
         run: |
           # Just a simple run to test the build script, no need to upload the generated iso!
           echo "TEST" > config.yaml
-          curl -fsSL https://raw.githubusercontent.com/rancher/elemental/main/elemental-iso-build -o elemental-iso-build
-          chmod +x ./elemental-iso-build
```

Comment:
@Itxaka this logic we done from some specific purpose I can't see? This prevents testing script changes on PRs since it always pulls the script from `main`, it does not execute the script form the PR.

I am wondering if this was done for some specific reason I can't see or it is just a leftover.

Reply:
That job is not there to prevent broken scripts getting in, it was just added as a extra to test the validity of the script on main. If you check you will see that it doesn't even checkout the source code, so its like an external user getting the script. 

Please add a different script to check the validity of the modified script instead as it has a different scope.

Reply:
Finally kept as it was